job-runner/chain/chain.go

```diff
@@ -38,10 +48,44 @@ func NewChain(jc *proto.JobChain) *chain {
 		jcMux:    &sync.RWMutex{},
 		jobChain: jc,
 
-		runningMux:         &sync.RWMutex{},
-		running:            map[string]proto.JobStatus{},
-		n:                  0,
-		sequenceRetryCount: sequenceRetryCount,
+		runningMux:     &sync.RWMutex{},
+		running:        map[string]proto.JobStatus{},
+		n:              0,
+		sequenceTries:  make(map[string]uint),
+		seqTriesMux:    &sync.RWMutex{},
+		totalTries:     make(map[string]uint),
+		totalTriesMux:  &sync.RWMutex{},
+		latestRunTries: make(map[string]uint),
+		latestTriesMux: &sync.RWMutex{},
+	}
+}
+
+// ResumeChain takes a SuspendedJobChain proto (from the RM) and turns it into a
+// Chain that the JR can use.
+func ResumeChain(sjc *proto.SuspendedJobChain) *chain {
```

Comment:
All this logic is called in only 1 place with the goal of making a normal chain: `MakeFromSJC`. I think it makes more sense directly in that method than standing alone here. This code is basically just factory code to build a chain.